pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -276,10 +279,85 @@ func (a Adapter) waitForManifestDeployCompletion(applicationName string, gvr sch
 	}
 }
 
+// GetProtocol returns the protocol string
+func getProtocol(route routev1.Route) string {
+	if route.Spec.TLS != nil {
+		return "https"
+	}
+	return "http"
+}
+
+func pluraliseKind(gvkKind string) (kind string) {
+	// gvkKind is normally a singular noun and we need to have the kind as a plural
+	// i.e. Deploment => Deployments
+	//      Route => Routes
+	//      Ingress => Ingresses
+	kind = strings.ToLower(gvkKind + "s")
+	if strings.HasSuffix(gvkKind, "s") {
+		kind = strings.ToLower(gvkKind + "es")
+	}
+	return kind
+}
```

Comment:
OPTIONAL: could be put in utils

Reply:
@EnriqueL8 Which utils did you think this should go in. 
The Route bits are OCP specific so  should these just go in occlient utils.go?

Reply:
Moved